### ISSUE TITLE
Internal error when specifying `--build-dir`

### ISSUE THREAD

**Issue by segeljakt** [ocaml/dune#5498]
## Expected Behavior

Dune builds my dune project (https://github.com/cda-group/arc/tree/master/arc-lang) at the directory specified through `--build-dir`.

## Actual Behavior

Dune produces an ICE:
* Command output: https://gist.github.com/segeljakt/b59d3d29579373012d292f12b5d76f6f
* Log: https://gist.github.com/segeljakt/1388891321410768c1d651acb6ddbb8b

## Specifications

- Version of `dune` (output of `dune --version`): `3.0.2`
- Version of `ocaml` (output of `ocamlc --version`): `4.12.0`
- Operating system (distribution and version): macOS Monterey 12.1 (ARM)

**Comment by frej** [comment]
A short reproducer which won't require anyone to figure out how to build the complete arc-project, this just builds the Ocaml part:

```
git clone https://github.com/cda-group/arc.git
TOP=$(pwd)
eval $(opam env)
cd arc/arc-lang
mkdir -p $TOP/build/llvm-build/arc-lang/build
dune build --root $TOP/arc/arc-lang --build-dir $TOP/build/llvm-build/arc-lang/build src/main.exe
```

**Comment by emillon** [comment]
Hi,
This has been fixed in 9b8c6cb907d3516014450317304e4ef92853b04a (part of the next release after 3.0.3).
